### PR TITLE
Chore/GitHub Actions, including CodeQL analysis

### PR DESCRIPTION
**Pull request by stephengtuggy**
Thank you for submitting a pull request and becoming a contributor to the Vega Strike Core Engine.

Please answer the following:

Code Changes:
- [ ] Have the PR Validation Tests been run? See https://github.com/vegastrike/Vega-Strike-Engine-Source/wiki/Pull-Request-Validation
- [ ] This is a documentation change only
- [x] This is a CI/CD change only

Issues:
- Please list any related issues
Addresses #382 

Purpose:
- What is this pull request trying to do? Replace Travis CI with GitHub Actions for Continuous Integration. CD to come as a separate step, once I see if this part works or not.
- What release is this for? 0.7.0
- Is there a project or milestone we should apply this to? 0.7.x

### CODE REVIEW

**Comment by stephengtuggy** [comment]
I already had this put together from a couple of months ago, so I merged the latest 0.7.x into it, and threw it up as a PR. This GH Actions setup can and probably will change, but hopefully it's at least a start.

**Comment by stephengtuggy** [comment]
Note that the travis-ci **.com** build has actually completed successfully. The GH webhook doesn't recognize it, because it's looking for the build on .org still.

**Comment by stephengtuggy** [comment]
.com build results here: https://travis-ci.com/github/stephengtuggy/Vega-Strike-Engine-Source/builds/211728386

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-18.04
+
+    strategy:
+      fail-fast: false
+      matrix:
+        # Override automatic language detection by changing the below list
+        # Supported options are ['csharp', 'cpp', 'go', 'java', 'javascript', 'python']
+        language: ['cpp', 'python']
```
Comment (BenjamenMeyer) [inline .github/workflows/codeql-analysis.yml:27]: Do we need to specify "C" here too? (Or should we?)

**Review by BenjamenMeyer — APPROVED** [review]
Not very familiar yet with GH Action syntax but from what little I'm familiar with: LGTM

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (stephengtuggy) [reply .github/workflows/codeql-analysis.yml:27]: I'm not sure if CodeQL supports C per se. At any rate, that would only be for the code security analysis part. Building the code should take place regardless.

**Review by Loki1950 — COMMENTED** [review]
looks good @stephengtuggy

**Pull request merged**